examples/platform/linux/AppMain.cpp

```diff
@@ -122,6 +122,15 @@ int ChipLinuxAppInit(int argc, char ** argv)
     err = Platform::MemoryInit();
     SuccessOrExit(err);
 
+    err = DeviceLayer::PlatformMgr().InitChipStack();
+    SuccessOrExit(err);
+
+    err = GetSetupPayload(LinuxDeviceOptions::GetInstance().payload, rendezvousFlags);
+    SuccessOrExit(err);
+
+    err = ParseArguments(argc, argv, customOptions);
```

Comment:
@vivien-apple Needed to parse the arguments before initializing KVS, otherwise, can't use the KVS location specified on command line

Reply:
Can you tell me what is needed exactly ? Is it the `err = ParseArguments(argc, argv, customOptions);` line or the need to call `err = DeviceLayer::PlatformMgr().InitChipStack();` ?

The reason I'm asking is because I'm planning to use the initialized KVS inside `PlatformManager::InitChipStack` and that's the reason why I have moved it in between `MemoryInit` and `InitChipStack`. 
The `ParseArguments` line is fine to me, `InitChipStack` will makes it harder for me to update #14660

Reply:
So the issue is if `ParseArguments` comes after KVS init (lines 134-144), the check for `if (LinuxDeviceOptions::GetInstance().KVS == nullptr)` will always be true and we always end up using the default KVS path CHIP_CONFIG_KVS_PATH.

I initially tried to move `ParseArguments` up before KVS init but if called before `InitChipStack`, I have problems.

Reply:
Nevermind, maybe the error I had ran into before was related to another change I was experimenting with. Moved ParseArgument up before KVS init and things seem to be working fine.